bindgen/ir/ty.rs

```diff
@@ -1099,6 +1100,20 @@ impl Type {
                         // within the clang parsing.
                         TypeKind::Opaque
                     } else {
+                        // Check if this type definition is an alias to a pointer of a `const
+                        // struct` with the same name and add the `_ptr` suffix to it to avoid name
+                        // collisions.
+                        if !ctx.options().c_naming {
+                            if let Some(inner_name) = dbg!(inner_spelling
+                                .strip_prefix("const struct ")
```

Comment:
This seems not specific to pointers-to-`const` - wouldn't something like `typedef struct foo { ... } *foo;` have the same issue?